data/icon-overlay.js

```diff
@@ -0,0 +1,90 @@
+const host = window.location.host;
+
+// YOUTUBE HOME PAGE
+// TODO: listen for infinite scrollesque updates and apply this again
+// if necessary
+const ytHomeContainers = Array.from(document.querySelectorAll('#feed .yt-lockup-thumbnail'));
+if ((host.indexOf('youtube.com') > -1) && ytHomeContainers.length) {
+  ytHomeContainers.forEach(el => {
+    el.classList.add('minvid__overlay__wrapper');
+    const tmp = getTemplate();
+    tmp.addEventListener('click', function(ev) {
+      evNoop(ev);
+      const partialUrl = el.querySelector('.yt-uix-sessionlink').getAttribute('href');
```

Comment:
`querySelector` returns `null` if nothing's found, so the `.getAttribute` could throw...unless `el` is some special thing